modules/core/src/main/java/org/apache/ignite/spi/encryption/EncryptionSpi.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.spi.encryption;
+
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+import org.apache.ignite.IgniteException;
+import org.apache.ignite.spi.IgniteSpi;
+
+/**
+ * SPI provides encryption features for an Ignite.
+ */
+public interface EncryptionSpi extends IgniteSpi {
+    /**
+     * Returns master key digest.
+     * Should always return same digest for a same key.
+     * Digest used for a configuration consistency check.
+     *
+     * @return Master key digest.
+     */
+    byte[] masterKeyDigest();
+
+    /**
+     * Creates new key for an encryption/decryption of cache persistent data: pages, WAL records.
+     *
+     * @return Newly created encryption key.
+     * @throws IgniteException If key creation failed.
+     */
+    Serializable create() throws IgniteException;
```

Comment:
It is more obvious to name this method as createKey()

Reply:
I think it's a matter of taste.

Reply:
Not only a matter of taste. Usually, create() method is placed in a Factory and it creates an instance of a class. Factory name and return type imply object type created. Here is it not clear without JavaDoc what method should return, so I suggest to change its naming.